DISCO_H747I dualcore support

ABOSTM opened the pull request:
### Description
Add 2 targets for DISCO_H747I dualcore:
* DISCO_H747I      -> for CM7 core
* DISCO_H747I_CM4  -> for CM4 core

**Current restrictions:**
* TICKLESS deactivated
* DeepSleep not supported (DeepSleep wrapped to sleep)

Warning: use of the same IP (example I2C1) by both core at the same time is not prevented,
but is strongly not recommended.
Some Hardware Semaphore are use for common IP, to manage concurrent access by both cores: Flash, GPIO, RCC.

Warning: Drag and drop of binary to DISCO_H747I disk will flash CM7. 
               In order to flash CM4, one can use STM32 CubeProgrammer tool.
              Example of CubeProgrammer command line to flash CM4:
               `PATH="/c/Program Files/STMicroelectronics/STM32Cube/STM32CubeProgrammer/bin/":$PATH`
                `STM32_Programmer_CLI -c port=SWD mode=UR -w BUILD/DISCO_H747I_CM4/ARM/<binary file name>  0x8100000` 

### Tests executed:
  * On CM7 (with a blinky binary on CM4), all tests on all 3 toolchains: ARM, GCC_ARM and IAR
  * On CM4 (with a blinky binary on CM7), all tests on toolchain IAR and GCC_ARM
  With toolchain ARM: blinky on both core simultaneously

All tests are passed except:
  * tests-mbed_hal-sleep_manager
  * tests-mbed_hal-sleep

This is normal because DeepSleep is not yet supported.

ABOSTM committed:
Mbed patch of STM32cube for bootloader: use NVIC_FLASH_VECTOR_ADDRESS

ABOSTM committed:
SystemCoreClock should correspond to current core clock and not D1 clock.

ABOSTM committed:
Update HAL/LL EXTI to have default API applied on current core and nott CPU1

jeromecoutant reviewed (approved):
Regression tests on other targets done.

As explained in the description, CM4 support can not be executed with automatic CI,
but I saw manual tests executed and passed!

Thx @ABOSTM !

bulislaw commented on features/mbedtls/targets/TARGET_STM/aes_alt.c:
```diff
@@ -103,11 +103,19 @@ void mbedtls_aes_free(mbedtls_aes_context *ctx)
     if (ctx == NULL) {
         return;
     }
+#if defined(DUAL_CORE)
```
@Patater please review

ABOSTM replied:
You are right this is up to the user to put whatever on both cores, and to figure out how they could talk to each other.
The modification in this specific file aes_alt.c concerns the potential concurent access (by CM4 and CM7) to common resource (RCC registers). This is done through Hardware Semaphore.

bulislaw commented on targets/TARGET_STM/TARGET_STM32H7/TARGET_STM32H747xI/TARGET_DISCO_H747I/PeripheralPins.c:
```diff
@@ -198,14 +198,14 @@ MBED_WEAK const PinMap PinMap_I2C_SCL[] = {
-    {PA_2,       PWM_2,  STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM2, 3, 0)}, // TIM2_CH3 // Connected to ETH_MDIO
-    {PA_2_ALT0,  PWM_5,  STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF2_TIM5, 3, 0)}, // TIM5_CH3 // Connected to ETH_MDIO
-    {PA_2_ALT1,  PWM_15, STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF4_TIM15, 1, 0)}, // TIM15_CH1 // Connected to ETH_MDIO
+//    {PA_0,       PWM_2,  STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM2, 1, 0)}, // TIM2_CH1 // Connected to PMOD\#1- USART2_CTS_NSS
```
Dead code.

jeromecoutant replied:
Yes and no...
We used to keep all the lines in all targets .
As the PinMap table is weak, idea was to give the availability for customer to change easily the default choice.

ABOSTM replied:
From pin configuration point of view, PA_0 can be connected to TIM2_CH1.
Nevertheless TIM2 and TIM5 are reserved for us_ticker.
That is why thoses lines are commented out: it is nomore usable for PWM purpose.
but we keep those lines as reminder to show the capability.
It is done the same way for example on target NUCLEO_L4R5ZI_P.

bulislaw replied:
Makes sense.

bulislaw reviewed (commented):
As far as I can tell, this is sort of a basic support only? It's up to the user to put whatever code on both cores and figure out how do they talk to each other?

bulislaw reviewed (approved):
Would it benefit from a readme file in the target directory with description how to build and flash both images?

mark-edgeworth reviewed (approved):
Reviewed only from a tools PoV. Looks ok.

ABOSTM committed:
DISCO_H747I Dualcore support

Add 2 targets for DISCO_H747I dualcore:
* DISCO_H747I      -> for CM7 core
* DISCO_H747I_CM4  -> for CM4 core

Current restrictions:
* TICKLESS deactivated
* DeepSleep not supported (DeepSleep wrapped to sleep)

Warning: use of the same IP (example I2C1) by both core at the same time is not prevented,
but is strongly not recommended.
Some Hardware Semaphore are use for common IP, to manage concurrent access by both cores: Flash, GPIO, RCC.

Warning: Drag and drop of binary to DISCO_H747I will flash CM7.
         In order to flash CM4, one can use STM32 CubeProgrammer tool.
